### DISCUSSION TITLE
Record wav file

### DISCUSSION THREAD

**Discussion by coulisse** [pschatzmann/arduino-audiokit#11] [Q&A] [answered]
Hello,
is it possible to have an example for recording a wav from lin input and write it to micro SD?
I wrote it, but the sound recorded is very noisy.

Thanks in advance
Corrado.

**Comment by pschatzmann** [comment]
I think you would take the following example https://github.com/pschatzmann/arduino-audio-tools/wiki/Audio-Encoding and just replace the NoiseGenerator with a  I2SStream or a AudioKitStream...

**Reply by coulisse** [reply]
Hello thanks for your quick reply.

I see in the example the following line:
`audioFile = SD.open("/test/002.**mp3**", O_WRITE | O_CREAT);`

but previously audioFile is defined as a "WavEncoder"
`EncodedAudioStream out_stream(&audioFile, new WAVEncoder()); `

so "/test/002.mp3" would it be ""/test/002.**wav**"?

Thanks
Corrado

**Comment by pschatzmann** [comment, marked as answer]
Absolutely - I corrected the example. I originally used a mp3 encoder in the example but later switched it to WAV.
I must have overseen this...
